app/forms/participants/participant_validation_form.rb

```diff
@@ -45,6 +45,11 @@ def attributes
     def date_of_birth=(value)
       @date_of_birth_invalid = false
       @date_of_birth = ActiveRecord::Type::Date.new.cast(value)
+
+      if invalid_date?(value)
+        @date_of_birth_invalid = true
+        @date_of_birth = value
+      end
```

Comment:
I don't think you need `invalid_date?` here. I'm pretty sure that `ActiveRecord::Type::Date.new.cast` will return `nil` if the date is not valid.

Reply:
If you enter 31/9/1990, for example, it amends the date to 01/10/1990. `Date.new(1990/9/31)` does return nil. Not sure what ActiveRecord is doing differently but something about 1 day over how many is in that month puts it to the 1st of the next month.